TATB - 39 [BUG]: Wrap Dialog Data into one config object for more readability

PrabhatJTDeveloper opened the pull request:
- for wraping dialog data into one config object changed the modal component ts file.
    - in modal component html file use that object .
    - and the modal is opening in signout which is in header component and skip button which is in quiz component so changed this two components.
    - add one object signoutModal in dialogData.json file.

PrabhatJTDeveloper committed:
- TATB 39
    - for wraping dialog data into one config object changed the modal component ts file.
    - in modal component html file use that object .
    - and the modal is opening in signout which is in header component and skip button which is in quiz component so changed this two components.
    - add one object signoutModal in dialogData.json file.

PrabhatJTDeveloper committed:
TATB - 30: When click skip button , pop up will remain even after question change
 -Give camelCase to variables
 -Fixed bug when click skip button the popup also swipes on timer end
 -Removed a console error

PrabhatJTDeveloper committed:
TATB:30 :
  -reverted chagnes
  - removed unwated code

charmijtdev commented on src/app/component/header/header.component.html:
```diff
@@ -52,6 +52,7 @@
                 class="btn btn-primary"
                 data-bs-toggle="modal"
                 data-bs-target="#exampleModal1"
```
what is use for  data-bs-toggle="modal"
                data-bs-target="#exampleModal1"

PrabhatJTDeveloper replied:
i removed it as it is not affecting anything

charmijtdev commented on src/app/dialog-service/modal/modal/modal.component.ts:
```diff
@@ -1,23 +1,14 @@
   selector: 'app-modal',
   templateUrl: './modal.component.html',
-  styleUrls: ['./modal.component.scss']
+  styleUrls: ['./modal.component.scss'],
```
please remove extra comma

PrabhatJTDeveloper replied:
yes, i removed it.

PrabhatJTDeveloper committed:
- TATB 39
    - solved the conflicts
    - removed unused class and id from the button tag of header component html file.
    - added about Modal data to the dialogeData.json

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -46,8 +44,8 @@ export class LoginComponent implements OnInit, OnDestroy {
     this.createForm();
-    if (localStorage.getItem('registeruser')?.length) {
-      let data: any = localStorage.getItem('registeruser');
+    if (localStorage.getItem('registerUser')?.length) {
```
no need to check length as localstorage will return stringnot array

PrabhatJTDeveloper replied:
ok, i removed it.

charmijtdev commented on src/app/component/quiz/quiz.component.ts:
```diff
@@ -65,7 +65,7 @@ export class Quizcomponent implements OnInit, OnDestroy {
+    event.returnValue = false;
   }
 
   mapJsonData() {
```
change variable name to mapJsonData()

charmijtdev replied:
and also can we change variable name to like getQuestionData()

charmijtdev commented on src/app/dialog-service/dialog.service.ts:
```diff
@@ -3,7 +3,7 @@ import { Injectable } from '@angular/core';
 import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
 import { ModalComponent } from './modal/modal/modal.component';
```
please do code indentation

PrabhatJTDeveloper replied:
ok,  done the code indentation

charmijtdev commented on src/app/dialog-service/dialog.service.ts:
```diff
@@ -14,17 +14,15 @@ export class DialogService {
+  
   ) {}
 
   dialogData = {...dialogData};
```
if not in used then remove it

PrabhatJTDeveloper replied:
yes, it is not in use so i removed it.

charmijtdev commented on src/app/dialog-service/modal/modal/modal.component.html:
```diff
@@ -1,12 +1,74 @@
-<div class="modal-header">
+<ng-template
+  [ngIf]="this.configData.type !== 'aboutModel'"
```
instead if else here can we match conidtion inside the model  <div class="modal-body">
    <p>{{ configData?.label }}</p>
  </div>
  <div class="modal-footer">
    <button
      type="button"
      class="btn btn-outline-dark"
      (click)="activeModal.close(false)"
    >
      {{ configData?.no }}
    </button>
    <button
      type="button"
      class="btn btn-outline-dark"
      (click)="activeModal.close(true)"
    > here ? and show data according to conditions

PrabhatJTDeveloper replied:
yes, i changed it.

PrabhatJTDeveloper committed:
- TATB 39
     - removed ?.length from the login component ts file.
     - removed import of dilogData.json from dialog service
     - updated modal component html file.

PrabhatJTDeveloper committed:
TATB - 29 : Registration page should contain header (#42)

* Technical assistant : Implement header for all pages

* TATB - 29 : Move inline CSS to external SCSS file

* TATB - 29 : Remove inline css and Remove popup box model in header section

* TATB - 29 :  Fixed Merge Conflicts

---------

Co-authored-by: Dipesh Gandhi <dipesh_gandhi9@gmail.com>
Co-authored-by: Charmi Chauhan <88572890+charmijtdev@users.noreply.github.com>

PrabhatJTDeveloper committed:
TATB- 33 : login page registration page flow change functionality (#41)

* TATB-05-BUG-Add-login-link-to-user-registration-form-login-form-component

* TATB-05-BUG-Add-login-link-to-user-registartion-form-login-form-component

* TATB - 33 : Login page Registration page flow change functionality

* TATB - 33 : Login page Registration page flow change functionality

* TATB - 33 : Login page Registartion page flow change functionality - changes - Login component,Registration component

* TATB - 33 : Login page Registration page flow change functionality changes - app module, registration component,toast component

* TATB - 33 : Login page Registration page flow change functinality changes - login, registration components

PrabhatJTDeveloper committed:
TATB-  38 : remove auth service use to auth guard for authentication (#43)

* TATB - 38 : Remove Auth service use to Auth guard for authentication

* TATB - 38 : Remove Auth service use to Auth guard for authentication changes-auth guards,login,header and question service name to chnage authentication service.

* TATB - 38 : Remove Auth service use to Auth guard for authentication changes - auth guard,login component,header, authentication service.

* TATB - 38 : Remove Auth service use to Auth guard for authentication.  changes - result component and authentication service

PrabhatJTDeveloper committed:
TATB - 40 : change design for confirmation dialog box (#44)

* TATB - 40 : change design for confirmation dialog box.

* TATB - 40 : change design for confirmation dialog box changes- modal component , login

* TATB - 40 : change design for confirmation dialog box - change login and registration components

PrabhatJTDeveloper committed:
- TATB 39
  - change the mapJsonData() to getQuestionData() function name in quiz component ts file and in statquiz component ts file mapJsonData() to getInstruction().

PrabhatJTDeveloper committed:
TATB - 37 : Redesign Quiz Test Component (#45)

* - TATB-37 redesign the quiz test template
  - for that change the html and scss file of the quiz component .

* - TATB 37
   - removed unnecessary id and role in the button from the html file
   - removed unwanted css in scss file

* - TATB 37
   - removed unwanted id and class from carousel in html file
   - removed the css which is also used as boostrap class.

* - TATB 37 removed unused div in html file

* - TATB 37
  - make change in condition of hightlighted text in quiz.component.html file

* - TATB 37
   - give the boostrap class and remove from the css
   - and make change in html file

* - TATB 37
   - make changes in html and scss file.

* - TATB 37
   - removed "this" from the {{this.selectedQuizType}} in the quiz component html file.

* - TATB 37
  - removed "this" from the html file

PrabhatJTDeveloper committed:
- TATB 39
    - removed inline css and give the extenal css
    - added the userId in the login component ts file
    - removed unnecassary line space and comma from the dialog service ts file
    - formated the code in modal component html file and also removed commented code.

PrabhatJTDeveloper committed:
- TATB 39
    - Removed extra comma from the login and header component ts file and also change the function name for opening dialog modal in header component ts file.
    - change the class name in header component scss file.
    - make change in dialogData.json file

PrabhatJTDeveloper committed:
TATB - 39 [BUG]: Wrap Dialog Data into one config object for more readability (#46)

* - TATB 39
    - for wraping dialog data into one config object changed the modal component ts file.
    - in modal component html file use that object .
    - and the modal is opening in signout which is in header component and skip button which is in quiz component so changed this two components.
    - add one object signoutModal in dialogData.json file.

* - TATB 39
     - removed ?.length from the login component ts file.
     - removed import of dilogData.json from dialog service
     - updated modal component html file.

* - TATB 39
    - removed inline css and give the extenal css
    - added the userId in the login component ts file
    - removed unnecassary line space and comma from the dialog service ts file
    - formated the code in modal component html file and also removed commented code.

* - TATB 39
    - Removed extra comma from the login and header component ts file and also change the function name for opening dialog modal in header component ts file.
    - change the class name in header component scss file.
    - make change in dialogData.json file

PrabhatJTDeveloper committed:
- TATB 39
   - Modify dialogData.json
   - make code documentation in scss file.
   - as change the json file also updated the html file.

charmijtdev committed:
removed confiict

charmijtdev committed:
Merge branch 'feature/development' into bug/TATB-39-BUG-wrap-dialog-data-into-one-config-object
